### ISSUE TITLE
Failure in detecting latest ffmpeg git master at compile time

### ISSUE THREAD

**Issue by dbermond** [mpv-player/mpv#4977]
mpv fails to compile with the latest ffmpeg git master. Maybe because ffmpeg 3.4 was just [branched out](http://git.videolan.org/?p=ffmpeg.git;a=shortlog;h=refs/heads/release/3.4)?

It compiles with fine until ffmpeg git master commit [58143b15adda6391ec07f3eb19e80ed91d801edd](http://git.videolan.org/?p=ffmpeg.git;a=commit;h=58143b15adda6391ec07f3eb19e80ed91d801edd).

From ffmpeg git master commit [6dfcbd80ad446ff163b47f2bf432bbf706436ea8](http://git.videolan.org/?p=ffmpeg.git;a=commit;h=6dfcbd80ad446ff163b47f2bf432bbf706436ea8) and later, it fails to compile with the following message at configure stage:
```
Checking for FFmpeg/Libav present                                    : yes 
Checking for libav* is FFmpeg                                        : no 
Checking for libav* is Libav                                         : no 
Checking for Libav/FFmpeg library versions                           : is_ffmpeg not found 
Unable to find development files for some of the required FFmpeg/Libav libraries. You need at least FFmpeg 3.2.2 or Libav 12. Aborting.
```

### mpv version and platform
**OS:** Arch Linux x86_64
**mpv:** git master

### Reproduction steps
```
$ git clone https://github.com/mpv-player/mpv.git
$ cd mpv
$ ./bootstrap.py
$ ./waf configure
```

### Expected behavior
Compiles with latest ffmpeg git master.

### Actual behavior
Does not compile with latest ffmpeg git master.

### Log file
[config.log](https://paste.ee/p/diMcl)

### Sample files
Not applicable.

**Comment by wm4** [comment]
Can't reproduce. Can you retry with git master? And it it still fails, try with this patch applied: https://pastebin.com/cy8s8zn3

**Comment by wm4** [comment]
Also, your ffmpeg build flags might be helpful.

**Comment by dbermond** [comment]
Hi wm4. Thank you for the reply.

> Can you retry with git master?

Already using git master for both ffmpeg and mpv :)

> And it it still fails, try with this patch applied:
> https://pastebin.com/cy8s8zn3

Applying this patch on ffmpeg git master fixes mpv compilation.
mpv can correctly detect ffmpeg git master at configure time with this patch.

> Also, your ffmpeg build flags might be helpful.

By investigating further I discovered that the ffmpeg build flags interfered with this.

- mpv git master correctly detects ffmpeg git master when using these ffmpeg build flags:
    - [https://paste.ee/p/5Bgju](https://paste.ee/p/5Bgju)

- But using these ffmpeg git master build flags makes mpv git master fail to detect ffmpeg at build time, working only with the mentioned patch:
    - [https://paste.ee/p/BRmNo](https://paste.ee/p/BRmNo)

It is known that **ffmpeg** commit [6dfcbd80ad446ff163b47f2bf432bbf706436ea8](http://git.videolan.org/?p=ffmpeg.git;a=commit;h=6dfcbd80ad446ff163b47f2bf432bbf706436ea8) caused a [confirmed problem](http://trac.ffmpeg.org/ticket/6729) (which is already [fixed](http://git.videolan.org/?p=ffmpeg.git;a=commit;h=734ed38931103c523c42333633a315a0ce1b5eb2)). Maybe this commit have broken other things too?

**Comment by wm4** [comment]
Which flags exactly cause failures?

**Comment by dbermond** [comment]
Cannot tell (yet).

I think it would be need to remove one by one and test compilations at each removal.

This would take some time. I'll try my best to find out which one is causing this.

**Comment by dbermond** [comment]
This was just fixed in ffmpeg git master by commit [25bd2f4f368ac350d635aaa50860ee642233dcc9](http://git.videolan.org/?p=ffmpeg.git;a=commit;h=25bd2f4f368ac350d635aaa50860ee642233dcc9).

It is working nice and there is nothing to test anymore. I think this issue can be closed.

I was thinking that it was a mpv issue. Sorry if I opened it in the wrong place. Fortunately, both projects share developers.

Thank you wm4 for the attention and support.

**Comment by wm4** [comment]
So it's solved? Closing then, thanks for your testing.
